src/mds/Mutation.h

```diff
@@ -435,12 +441,11 @@ struct MDRequestImpl : public MutationImpl {
   // indicates how may retries of request have been made
   int retry = 0;
 
-  bool is_batch_head = false;
+  std::map<int, std::unique_ptr<BatchOp> > *batch_op_map = nullptr;
```

Comment:
Ownership is not clear to me. Should this be a `std::shared_ptr<std::map<...>>`?

Reply:
it's a pointer to either CInode::batch_ops or CDentry::batch_ops. can not be shared ptr